### DISCUSSION TITLE
waitForTransactionReceipt - Cannot mix BigInt and other types

### DISCUSSION THREAD

**Discussion by dapjeongneo** [wevm/viem#880] [Question]
Hi team, can someone pls take a look at my following code. Am i doing `waitForTransactionReceipt` wrongly? i get the following error but the transaction was successful. Thank you!
```
walletClient.writeContract(tx).then(async (hash) => {
  await publicClient
    .waitForTransactionReceipt({ hash: hash })
    .then((receipt) => {
      if (receipt.status === "success") {
        console.log("success");
      }
    })
    .catch((e) => {
      // tx failed:
      // TypeError: Cannot mix BigInt and other types, use explicit conversions
      // at Object.onBlockNumber (waitForTransactionReceipt.js:100:1)
      console.error("tx failed:", e)
    })
})
```

**Comment by jxom** [comment]
Hard to tell without knowing what runtime environment you are running in or what bundler you are using. Is this Node, Next.js, Create React App, Remix, Bun, etc? A minimal reproduction would help.
